idaes/models/unit_models/heat_exchanger_1D.py

```diff
@@ -430,39 +430,62 @@ def build(self):
 
         # Set flow directions for the control volume blocks and specify
         # discretization if not specified.
-        if self.config.flow_type == HeatExchangerFlowPattern.cocurrent:
-            set_direction_hot = FlowDirection.forward
-            set_direction_cold = FlowDirection.forward
+        if self.config.hot_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the hot side of the "
+                "co-current heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the hot side."
+            )
+            self.config.hot_side.transformation_method = "dae.finite_difference"
+        if self.config.cold_side.transformation_method is useDefault:
+            _log.warning(
+                "Discretization method was "
+                "not specified for the cold side of the "
+                "co-current heat exchanger. "
+                "Defaulting to finite "
+                "difference method on the cold side."
+            )
+            self.config.cold_side.transformation_method = "dae.finite_difference"
+
+        if (
+            self.config.hot_side.transformation_method
+            != self.config.cold_side.transformation_method
+        ):
+            raise ConfigurationError(
+                "HeatExchanger1D only supports similar transformation "
+                "methods on the hot and cold side domains for "
+                "both cocurrent and countercurrent flow patterns. "
+                f"Found method {self.config.hot_side.transformation_method} "
+                f"on hot side and method {self.config.cold_side.transformation_method} "
+                "on cold side."
+            )
+        if self.config.hot_side.transformation_method == "dae.collocation":
+            if (
+                self.config.hot_side.transformation_scheme is useDefault
+                or self.config.cold_side.transformation_scheme is useDefault
+            ):
+                raise ConfigurationError(
+                    "If a collocation method is used for HeatExchanger1D, the user "
+                    "must specify the transformation scheme they want to use."
+                )
             if (
-                self.config.hot_side.transformation_method
-                != self.config.cold_side.transformation_method
-            ) or (
                 self.config.hot_side.transformation_scheme
```

Comment:
As you mentioned, this is only caught if collocation is used. Should this check/raise be included for finite difference as well?

Reply:
I catch it for collocation because those methods create additional points within finite elements. If the points don't match up (which they won't for different schemes), you'll get errors trying to create heat linking equations between the control volumes.

For finite differences, no additional points are created, so there's no need to check that the schemes are the same (except to perhaps warn the user that energy may not be conserved).